### DISCUSSION TITLE
Consultation inquiry

### DISCUSSION THREAD

**Discussion by jhelvy** [rbind/yihui#1562] [General]
Yihui,

First, let me say that I have very long time admirer of all the things you have done. So much of my professional work is associated with the software you've built. I was a pretty early adopter of {rmarkdown}, and all of my classes I teach are built using it. I'm also am a pretty heavy user of {xaringan}, and I built the [{renderthis}](https://github.com/jhelvy/renderthis) package along with Garrick to help render xaringan slides to different formats, like pdf. Thank you so much for what you've built and for all the other help you've offered the broader R community! 🙏 

I read your blog post about leaving posit, and I know it's probably not been a great time the last few months. I really hope something will come through and things will work out soon. In the meantime, I was wondering if you would be willing to help me with an idea as a paid consultant. Basically, I want to schedule a call to toss around some ideas. I would pay you for your time, and I think it would really help me make a decision about whether I should pursue this project I'm thinking about (and if so, maybe some directions to consider).

The gist is that I want to build a rmarkdown-based survey platform. I actually wrote a [blog post](https://www.jhelvy.com/posts/2023-04-06-markdown-surveys/) a while back about this idea, but I haven't pursued it further. It would be something like {xaringan} in terms of the user interface (all content defined in a single .Rmd file), but it would render to a survey app instead of a slide deck. Making the app live is an open question, but I was thinking that rendering to a shiny app might be feasible, with something really simple like a Google sheet for a back end to store responses. So {rmarkdown} + {shiny} + {googlesheets4} + {surveydown} (the package I want to build).

I feel like all the pieces are there to make this, but every time I've started down a path I hit dead ends where I just don't know enough about how to pull all the pieces together. That's why I'm reaching out. I was thinking you might have some ideas of how to maybe architect this. I'm not asking that you get involved in development (unless you want to of course), just more thoughts on whether this might be feasible and maybe any advice on architecture. Let me know what you think.

Cheers,
JP

**Comment by yihui** [comment]
Hi John, thanks a lot for offering this opportunity, and sorry for the long delay to get back to you. I know {surveydown} has already been born (congrats!), so I guess you won't need much help from me any more (LLMs are likely to be better than me now). That said, HTML/CSS/JS are still my favorite things to play with. I'll be happy to contribute a little bit if your team is stuck in this area. I spent about half an hour touring around the documentation website and Github repo. It looks everything is going pretty well. I just posted a quick reply to https://github.com/surveydown-dev/surveydown/issues/260 and hope it helps (of course, no need to pay me for that trivial reply).

**Reply by jhelvy** [reply]
Thanks so much! Yeah, we spent the summer of 2024 pulling together some of the basic ideas that eventually became the architecture we now have. Things have moved quickly with the help of Claude to get around some bottlenecks. The platform is working surprisingly well for such a hybrid architecture of using Quarto to define the UI of a shiny app. Still, there are some aspects that are a bit odd about it. I think people need to be pretty comfortable with both Quarto / Rmd and shiny to really leverage it to its fullest potential for really complex survey designs. We'd certainly welcome any ideas / suggestions you have!
